<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>prora: Unifying methods on gene set enrichment • prora</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="prora: Unifying methods on gene set enrichment">
<meta property="og:description" content="prora">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">prora</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/vignette.html">prora: Unifying methods on gene set enrichment</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="vignette_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>prora: Unifying methods on gene set enrichment</h1>
                        <h4 class="author">Lucas Heinrich Kook</h4>
            <address class="author_afil">
      University of Zurich<br><h4 class="author">Witold Wolski</h4>
            <address class="author_afil">
      University of Zurich<br><h4 class="date">28 January 2021</h4>
      
      
      <div class="hidden name"><code>vignette.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div id="abstract" class="section level1">
<h1 class="hasAnchor">
<a href="#abstract" class="anchor"></a>Abstract</h1>
<p>A plethora of R packages exist on CRAN and Bioconductor to perform over-representation analysis (ORA) and gene set enrichment analysis (GSEA). However, consistency in the underlying nomenclature for specific analyses and user friendly implementation is still lacking. <code>prora</code> aims at unifying ID mapping and enrichment analysis in a syntactically coherent and intuitive way, while ensuring reproducibility of results. <code>prora</code> primarily consists of wrapper functions around the <em><a href="https://CRAN.R-project.org/package=sigora">sigora</a></em> and <em><a href="https://CRAN.R-project.org/package=WebGestaltR">WebGestaltR</a></em> packages from CRAN and <em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> based reports for visualisation and contextualisation of analysis results.</p>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Methods to identify differentially regulated pathways among a vast background of quantified genes or proteins differ in complexity of the underlying algorithms and computations, ranging from the application of Fisher’s exact test in ORA and extensions to gene pairs <span class="citation">(Foroushani, Brinkman, and Lynn 2013)</span> to more complex statistical procdures in gene set enrichment analysis <span class="citation">(Subramanian et al. 2005)</span> and network based procedures <span class="citation">(Glaab et al. 2012)</span>. Almost all methods come with their own implementation as an <em>R</em> package, <em>i.e.</em> ORA in both <em><a href="https://CRAN.R-project.org/package=sigora">sigora</a></em> and <em><a href="https://CRAN.R-project.org/package=WebGestaltR">WebGestaltR</a></em>, network based gene enrichment analysis in <em><a href="https://CRAN.R-project.org/package=enrichnet">enrichnet</a></em> and the list keeps going on. Naturally, different implementations come with different – and often inconsistent – features, such as ID mapping or demand a lot of knowledge from the user in setting up appropriate data structures and other preconditions. <code>prora</code> aims at providing a consistent and user friendly interface, which will be introduced in the following section. Elaborating on the methodological and statistical aspects of the underlying methods is beyond the scope of this vignette and the reader is referred to the publications in the reference section.</p>
</div>
<div id="installation-and-loading" class="section level1">
<h1 class="hasAnchor">
<a href="#installation-and-loading" class="anchor"></a>Installation and loading</h1>
<p>After installing the packages from <em>Bioconductor</em> as shown below, the package can be loaded using <code><a href="https://rdrr.io/r/base/library.html">library(prora)</a></code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"prora"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">prora</span><span class="op">)</span></code></pre></div>
</div>
<div id="example-workflow" class="section level1">
<h1 class="hasAnchor">
<a href="#example-workflow" class="anchor"></a>Example workflow</h1>
<p>We shall analyse proteomic data with the following format:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"exampleContrastData"</span><span class="op">)</span>
<span class="co">#glimpse(exampleContrastData)</span></code></pre></div>
<p>The first column <code>protein_Id</code> contains FASTA formatted human protein accessions and the second column fold change estimates. <em>e.g.</em> as obtained by a <code><a href="https://rdrr.io/pkg/limma/man/toptable.html">limma::toptable()</a></code> call. The first step is now to translate the FASTA headers into Uniprot accessions for subsequent ID mapping.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dd</span> <span class="op">&lt;-</span> <span class="fu">prora</span><span class="fu">::</span><span class="fu"><a href="../reference/get_UniprotID_from_fasta_header.html">get_UniprotID_from_fasta_header</a></span><span class="op">(</span><span class="va">exampleContrastData</span><span class="op">)</span>
<span class="co">#glimpse(dd)</span></code></pre></div>
<p>We see that we have lost 0 accessions which did not correspond to a FASTA format. Now we apply ORA, sigORA and GSEA to the obtained ranked protein list. To apply sigORA and ORA one must first compute the background GPS repository using <code><a href="../reference/makeGPS_wrappR.html">makeGPS_wrappR()</a></code>. The advantage is that backgrounds can be specified separately for each experiment or combined for several batches of experiments. However, computing the accession pair signatures and calculating their respective weights is computationally intensive and potentially time consuming.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"idmap"</span>, package <span class="op">=</span> <span class="st">"sigora"</span><span class="op">)</span>
<span class="va">myGPSrepo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeGPS_wrappR.html">makeGPS_wrappR</a></span><span class="op">(</span>ids <span class="op">=</span> <span class="va">dd</span><span class="op">$</span><span class="va">UniprotID</span>, target <span class="op">=</span> <span class="st">"GO"</span><span class="op">)</span>


<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sigoraWrappR.html">sigoraWrappR</a></span><span class="op">(</span>data <span class="op">=</span>  <span class="va">dd</span>,
                    threshold <span class="op">=</span> <span class="fl">0.2</span>,
                    score_col <span class="op">=</span> <span class="st">"estimate"</span>,
                    GPSrepos <span class="op">=</span> <span class="va">myGPSrepo</span><span class="op">$</span><span class="va">gps</span>,
                    greater_than <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>The result of <code><a href="../reference/sigoraWrappR.html">sigoraWrappR()</a></code> is a list containing the sigORA and ORA results along with user specified inputs that will be used in the reports described in section @ref(reports).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "sigora"                 "ora"                    "threshold"             
## [4] "GPSrepository"          "database"               "data"                  
## [7] "proteinsAfterFiltering" "direction"</code></pre>
<p>Under the hood a lot of ID mapping has taken place of which the user is unaware. To check the ID mapping efficiency when mapping Uniprot IDs to GO, KEGG or ENTREZ accessions the package comes with a <code><a href="../reference/checkIDmappingEfficiency.html">checkIDmappingEfficiency()</a></code> function to check the number of lost IDs.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/checkIDmappingEfficiency.html">checkIDmappingEfficiency</a></span><span class="op">(</span><span class="va">dd</span><span class="op">$</span><span class="va">UniprotID</span>, keytype <span class="op">=</span> <span class="st">"UNIPROT"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>##                InputIDs ENTREZIDs KEGGIDs  GOIDs
## Number             3701   3680.00 1534.00 3653.0
## Percent Mapped      100     99.43   41.45   98.7</code></pre>
<p>Evidently, the mapping from Uniprot to KEGG IDs seems to be fairly inefficient and results should be interpreted with care.</p>
</div>
<div id="reports" class="section level1">
<h1 class="hasAnchor">
<a href="#reports" class="anchor"></a>Reports</h1>
<p>Good reporting and reproducibility of all intermediate and final results is key to good scientific practice. In this spirit, <code>prora</code> provides a reproducible, <em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> based reports for each method. To highlight the most important targets, settings and parameters, a walkthrough is given in this part of the vignette.</p>
<p>To run all methods and produce the reports on the example data from the prior section, the following parameters must be set in advance.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">organism</span> <span class="op">&lt;-</span> <span class="st">"hsapiens"</span> <span class="co"># Organism to which the FASTA headers correspond</span>
<span class="va">ID_col</span> <span class="op">&lt;-</span> <span class="st">"UniprotID"</span> <span class="co"># Column containing the IDs</span>
<span class="va">fc_col</span> <span class="op">&lt;-</span> <span class="st">"estimate"</span> <span class="co"># Column containing the estimates (in this case log fold changes)</span>
<span class="va">target_GSEA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="co"># all possible targets for GSEA</span>
  <span class="st">"geneontology_Biological_Process"</span>,
  <span class="st">"geneontology_Cellular_Component"</span>,
  <span class="st">"geneontology_Molecular_Function"</span>
<span class="op">)</span>
<span class="va">target_SIGORA</span> <span class="op">&lt;-</span> <span class="co"># all possible targets for sigORA</span>
  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GO"</span>, <span class="st">"KEGG"</span>, <span class="st">"reactome"</span><span class="op">)</span>
<span class="va">fc_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.5</span> <span class="co"># threshold for ORA methods</span>
<span class="va">greater</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span> <span class="co"># direction of the threshold</span>
<span class="va">nperm</span> <span class="op">&lt;-</span> <span class="fl">10</span> <span class="co"># number of permutations used to compute enrichment scores in GSEA</span>
<span class="va">odir</span> <span class="op">&lt;-</span> <span class="st">"tmp"</span> <span class="co"># output directory to write all reports</span></code></pre></div>
<p>To iterate over all possible target directly, the following workflow functions can be used. They take the above specified parameters as inputs and facilitate the enrichment analysis workflows tremendously. The results will be saved to the specified output directory with a subfolder for each method and target.</p>
<p>The following code chunk iterates the <code>runGSEA()</code> function over all possible GSEA target databases. Internally, <code>runGSEA()</code> calls <code>WebGestaltR()</code> for the analysis and compiles <code>prora's</code> in the same output directory.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">target_GSEA</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="../reference/runWebGestaltGSEA.html">runWebGestaltGSEA</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="va">dd</span>,
    fpath <span class="op">=</span> <span class="st">""</span>,
    ID_col <span class="op">=</span> <span class="va">ID_col</span>,
    score_col <span class="op">=</span> <span class="va">fc_col</span>,
    organism <span class="op">=</span> <span class="va">organism</span>,
    target <span class="op">=</span> <span class="va">x</span>,
    nperm <span class="op">=</span> <span class="va">nperm</span>,
    outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">odir</span>, <span class="st">"WebGestaltGSEA"</span><span class="op">)</span>
  <span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<p>Analogously, <code><a href="../reference/runWebGestaltORA.html">runWebGestaltORA()</a></code> and <code><a href="../reference/runSIGORA.html">runSIGORA()</a></code> embody a whole workflow and save nicely formatted results to the output directory.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">target_GSEA</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="../reference/runWebGestaltORA.html">runWebGestaltORA</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="va">dd</span>,
    fpath <span class="op">=</span> <span class="st">""</span>,
    organism <span class="op">=</span> <span class="va">organism</span>,
    ID_col <span class="op">=</span> <span class="va">ID_col</span>,
    target <span class="op">=</span> <span class="va">x</span>,
    threshold <span class="op">=</span> <span class="va">fc_threshold</span>,
    greater <span class="op">=</span> <span class="va">greater</span>,
    nperm <span class="op">=</span> <span class="va">nperm</span>,
    score_col <span class="op">=</span> <span class="va">fc_col</span>,
    outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">odir</span>, <span class="st">"WebGestaltORA"</span><span class="op">)</span>
  <span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<p><code><a href="../reference/runSIGORA.html">runSIGORA()</a></code> calls the <code><a href="../reference/sigoraWrappR.html">sigoraWrappR()</a></code> function which was introduced in section @ref(Example workflow) and compiles the built-in report.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">target_SIGORA</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="../reference/runSIGORA.html">runSIGORA</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="va">dd</span>,
    target <span class="op">=</span> <span class="va">x</span>,
    score_col <span class="op">=</span> <span class="va">fc_col</span>,
    threshold <span class="op">=</span> <span class="va">fc_threshold</span>,
    greater <span class="op">=</span> <span class="va">greater</span>,
    outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">odir</span>, <span class="st">"sigORA"</span><span class="op">)</span>
  <span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<p>In the following the <code>runGSEA()</code> function’s inner workings will be discussed in more detail. After creating the output directory <code>outdir</code> the provided data is fed into <em><a href="https://CRAN.R-project.org/package=WebGestaltR">WebGestaltR</a></em>’s <code>WebGestaltR(..., enrichMethod = "GSEA")</code>. The results are then used to generate the report provided by the <code>prora</code> package<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>. All <code>WebGestaltR</code> and <code>prora</code> results can then be found in the output directory.</p>
</div>
<div id="command-line-tools" class="section level1">
<h1 class="hasAnchor">
<a href="#command-line-tools" class="anchor"></a>Command line tools</h1>
<p>The enrichment methods in this package (ORA, GSEA sigORA) come with a <em><a href="https://CRAN.R-project.org/package=docopt">docopt</a></em> based command line tool to facilitate analysing batches of files<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>. The main advantage of docopt is that all tools have to come with a thorough documentation where it is also possible to provide default values. For running GSEA, the command line tool is:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="st">"WebGestaltR GSEA for multigroup reports

Usage:
  lfq_multigroup_gsea.R &lt;grp2file&gt; [--organism=&lt;organism&gt;] [--outdir=&lt;outdir&gt;] [--idtype=&lt;idtype&gt;] [--ID_col=&lt;ID_col&gt;]  [--nperm=&lt;nperm&gt;] [--score_col=&lt;score_col&gt;] [--contrast=&lt;contrast&gt;]

Options:
  -o --organism=&lt;organism&gt; organism [default: hsapiens]
  -r --outdir=&lt;outdir&gt; output directory [default: results_gsea]
  -t --idtype=&lt;idtype&gt; type of id used for mapping [default: uniprotswissprot]
  -i --ID_col=&lt;ID_col&gt; Column containing the UniprotIDs [default: UniprotID]
  -n --nperm=&lt;nperm&gt; number of permutations to calculate enrichment scores [default: 500]
  -e --score_col=&lt;score_col&gt; column containing fold changes [default: pseudo_estimate]
  -c --contrast=&lt;contrast&gt; column containing fold changes [default: contrast]

Arguments:
  grp2file  input file
"</span> <span class="op">-&gt;</span> <span class="va">doc</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/docopt/docopt.R">docopt</a></span><span class="op">)</span>

<span class="va">opt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/docopt/man/docopt.html">docopt</a></span><span class="op">(</span><span class="va">doc</span><span class="op">)</span></code></pre></div>
<p>It can be run using the following command in a UNIX/Linux terminal, specifying the organism as mouse:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1"></a><span class="ex">Rscript</span> path/to/lfq_multigroup_gsea.R path/to/group2file.xlsx -o mmusculus</span></code></pre></div>
<p>The help file can also be accessed via the command line using:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1"></a><span class="ex">Rscript</span> path/to/lfq_multigroup_gsea.R -h</span></code></pre></div>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 19041)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_United Kingdom.1252 
## [2] LC_CTYPE=English_United Kingdom.1252   
## [3] LC_MONETARY=English_United Kingdom.1252
## [4] LC_NUMERIC=C                           
## [5] LC_TIME=English_United Kingdom.1252    
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] prora_0.1.0       WebGestaltR_0.4.4 forcats_0.5.0     stringr_1.4.0    
##  [5] dplyr_1.0.3       purrr_0.3.4       readr_1.4.0       tidyr_1.1.2      
##  [9] tibble_3.0.5      ggplot2_3.3.3     tidyverse_1.3.0   BiocStyle_2.16.0 
## 
## loaded via a namespace (and not attached):
##  [1] fs_1.5.0             lubridate_1.7.9.2    bit64_4.0.5         
##  [4] doParallel_1.0.16    httr_1.4.2           UpSetR_1.4.0        
##  [7] rprojroot_2.0.2      tools_4.0.3          backports_1.2.1     
## [10] doRNG_1.8.2          R6_2.5.0             BiocGenerics_0.34.0 
## [13] DBI_1.1.1            colorspace_2.0-0     withr_2.4.1         
## [16] gridExtra_2.3        tidyselect_1.1.0     bit_4.0.4           
## [19] compiler_4.0.3       Biobase_2.48.0       textshaping_0.2.1   
## [22] cli_2.2.0            rvest_0.3.6          xml2_1.3.2          
## [25] desc_1.2.0           slam_0.1-48          scales_1.1.1        
## [28] sigora_3.0.5         pkgdown_1.6.1        systemfonts_0.3.2   
## [31] apcluster_1.4.8      digest_0.6.27        rmarkdown_2.6       
## [34] svglite_1.2.3.2      pkgconfig_2.0.3      htmltools_0.5.1.1   
## [37] dbplyr_2.0.0         rlang_0.4.10         readxl_1.3.1        
## [40] rstudioapi_0.13      RSQLite_2.2.3        generics_0.1.0      
## [43] jsonlite_1.7.2       magrittr_2.0.1       GO.db_3.11.4        
## [46] Matrix_1.2-18        S4Vectors_0.26.1     Rcpp_1.0.6          
## [49] munsell_0.5.0        fansi_0.4.2          gdtools_0.2.3       
## [52] lifecycle_0.2.0      stringi_1.5.3        whisker_0.4         
## [55] yaml_2.2.1           org.Hs.eg.db_3.11.4  plyr_1.8.6          
## [58] blob_1.2.1           grid_4.0.3           parallel_4.0.3      
## [61] crayon_1.3.4         lattice_0.20-41      haven_2.3.1         
## [64] hms_1.0.0            knitr_1.30           ps_1.5.0            
## [67] pillar_1.4.7         igraph_1.2.6         rngtools_1.5        
## [70] codetools_0.2-16     stats4_4.0.3         reprex_0.3.0        
## [73] glue_1.4.2           evaluate_0.14        BiocManager_1.30.10 
## [76] modelr_0.1.8         vctrs_0.3.6          foreach_1.5.1       
## [79] cellranger_1.1.0     gtable_0.3.0         assertthat_0.2.1    
## [82] xfun_0.20            broom_0.7.3          reactome.db_1.70.0  
## [85] ragg_0.4.1           iterators_1.0.13     IRanges_2.22.2      
## [88] AnnotationDbi_1.50.3 memoise_1.1.0        ellipsis_0.3.1</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Foroushani2013">
<p>Foroushani, Amir B. K., Fiona S. L. Brinkman, and David J. Lynn. 2013. “Pathway-Gps and Sigora: Identifying Relevant Pathways Based on the over-Representation of Their Gene-Pair Signatures.” <em>PeerJ</em> 1 (December): e229. <a href="https://doi.org/10.7717/peerj.229">https://doi.org/10.7717/peerj.229</a>.</p>
</div>
<div id="ref-glaab2012enrichnet">
<p>Glaab, Enrico, Anais Baudot, Natalio Krasnogor, Reinhard Schneider, and Alfonso Valencia. 2012. “EnrichNet: Network-Based Gene Set Enrichment Analysis.” <em>Bioinformatics</em> 28 (18): i451–i457.</p>
</div>
<div id="ref-Subramanian15545">
<p>Subramanian, Aravind, Pablo Tamayo, Vamsi K. Mootha, Sayan Mukherjee, Benjamin L. Ebert, Michael A. Gillette, Amanda Paulovich, et al. 2005. “Gene Set Enrichment Analysis: A Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles.” <em>Proceedings of the National Academy of Sciences</em> 102 (43): 15545–50. <a href="https://doi.org/10.1073/pnas.0506580102">https://doi.org/10.1073/pnas.0506580102</a>.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>The source files for all reports can be found in <code><a href="https://rdrr.io/r/base/system.file.html">system.file("rmarkdown_reports", package = "fgczgseaora")</a></code><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>Those scripts can be found in <code><a href="https://rdrr.io/r/base/system.file.html">system.file("run_scripts", package = "prora")</a></code><a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Lucas Kook, Witold E. Wolski.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
